### ISSUE TITLE
Update basic auth filter to consume base64 encoded password

### ISSUE THREAD

**Issue by bianpengyuan** [istio-ecosystem/wasm-extensions#11]
Currently basic auth filter only consumes plain text `username:passwd`. This is to track the work to make it read base64 encoded username:passwd instead.

**Comment by mboveri** [comment]
Would this also add functionality to allow the basic auth filter to be able to consume from a Kubernetes secret or environment variable generated by a Kubernetes secret?

**Comment by bianpengyuan** [comment]
Yeah it depends on this: https://github.com/proxy-wasm/proxy-wasm-cpp-host/issues/127

**Comment by bianpengyuan** [comment]
And envoy just added support for Wasm env variable read: https://github.com/envoyproxy/envoy/issues/14958 I can add some support based on that.

**Comment by mboveri** [comment]
That would be great!

**Comment by johnzheng1975** [comment]
@bianpengyuan , had istio supported this -- wasm can read env variable from istio-proxy now? Can you provide some example. Thanks.

**Comment by johnzheng1975** [comment]
> And envoy just added support for Wasm env variable read: [envoyproxy/envoy#14958](https://github.com/envoyproxy/envoy/issues/14958) I can add some support based on that.

@bianpengyuan , had istio supported this -- wasm can read env variable from istio-proxy now? Can you provide some example. Thanks.
